### Excercise 1.1

---

The above gives about a 1.5% chance of randomly selecting the correct cups. Calculate the minimum number of cups of tea required to ensure the probability of randomly selecting the correct cups is less than or equal to 1%.


The following is adapted from: https://stackoverflow.com/a/4941932

In [11]:
#Math Module
import math

In [12]:
#No of ways selecting 4 cups from 8
math.comb(8,4)

70

In [13]:
#No of ways selecting 5 cups from 10
math.comb(10,5)

252

*As you can see from above there are 252 possible outcomes*

In [14]:
#Probability of randomoly selecting correct 5 cups is:
1.0 / math.comb(10,5)

0.003968253968253968

So if the experiment is done with 10 cups, five with milk first and five with tea first, then selecting correctly randomly is less than 1%

In [15]:
#No of ways selecting 6 cups from 12
math.comb(12,6)

924

In [16]:
#Probability of randomoly selecting correct 6 cups is:
1.0 / math.comb(12,6)

0.0010822510822510823

This too is also less than 1%

Bonus: How many would be required if you were to let the taster get one cup wrong while maintaining the 1% threshold?

### Excercise 1.2

---

Use scipy's version of Fisher's exact test to simulate the Lady Tasting Tea problem.

In [18]:
#import the relevant packages
import scipy.stats as ss

Code was taken and adapted from: https://towardsdatascience.com/fishers-exact-test-from-scratch-with-python-2b907f29e593

### Excercise 2.1

---
Take the code from the Examples section of the scipy stats documentation for independent samples t-tests, add it to your own notebook and add explain how it works using MarkDown cells and code comments. Improve it in any way you think it could be improved.